sympy/algebras/quaternion.py

```diff
@@ -78,8 +94,13 @@ class Quaternion(Expr):
 
     is_commutative = False
 
-    def __new__(cls, a=0, b=0, c=0, d=0, real_field=True):
-        a, b, c, d = map(sympify, (a, b, c, d))
+    def __new__(cls, a=0, b=0, c=0, d=0, real_field=True, norm=None):
+        a, b, c, d, norm = map(sympify, (a, b, c, d, norm))
+
+        if (norm is not None and
```

Comment:
Or if norm is 0, that is a problem,  too.

Reply:
If `norm` is zero then it would indeed (probably) be a useless quaternion, but is this something we would like to avoid completely?
What if I add a `UserWarning` instead?